[BUG] transform fuse for thinruntime with err: elements differ

**What is your environment(Kubernetes version, Fluid version, etc.)**

**Describe the bug**

```bash
--- FAIL: TestThinEngine_transformFuse (0.01s)
    --- FAIL: TestThinEngine_transformFuse/test (0.01s)
        deepequal.go:27: 
            	Error Trace:	/home/runner/work/fluid/fluid/src/github.com/fluid-cloudnative/fluid/pkg/ddc/thin/deepequal.go:27
            	            				/home/runner/work/fluid/fluid/src/github.com/fluid-cloudnative/fluid/pkg/ddc/thin/deepequal.go:35
            	            				/home/runner/work/fluid/fluid/src/github.com/fluid-cloudnative/fluid/pkg/ddc/thin/transform_fuse_test.go:517
            	Error:      	elements differ
            	            	
            	            	extra elements in list A:
            	            	([]interface {}) (len=1) {
            	            	 (v1.EnvVar) {
            	            	  Name: (string) (len=13) "MOUNT_OPTIONS",
            	            	  Value: (string) (len=7) "a=b,c=d",
            	            	  ValueFrom: (*v1.EnvVarSource)(<nil>)
            	            	 }
            	            	}
            	            	
            	            	
            	            	extra elements in list B:
            	            	([]interface {}) (len=1) {
            	            	 (v1.EnvVar) {
            	            	  Name: (string) (len=13) "MOUNT_OPTIONS",
            	            	  Value: (string) (len=7) "c=d,a=b",
            	            	  ValueFrom: (*v1.EnvVarSource)(<nil>)
            	            	 }
            	            	}
            	            	
            	            	
            	            	listA:
            	            	([]v1.EnvVar) (len=4) {
            	            	 (v1.EnvVar) {
            	            	  Name: (string) (len=1) "a",
            	            	  Value: (string) (len=1) "b",
            	            	  ValueFrom: (*v1.EnvVarSource)(<nil>)
            	            	 },
            	            	 (v1.EnvVar) {
            	            	  Name: (string) (len=1) "b",
            	            	  Value: (string) "",
            	            	  ValueFrom: (*v1.EnvVarSource)({
            	            	   FieldRef: (*v1.ObjectFieldSelector)(<nil>),
            	            	   ResourceFieldRef: (*v1.ResourceFieldSelector)(<nil>),
            	            	   ConfigMapKeyRef: (*v1.ConfigMapKeySelector)({
            	            	    LocalObjectReference: (v1.LocalObjectReference) {
            	            	     Name: (string) (len=7) "test-cm"
            	            	    },
            	            	    Key: (string) "",
            	            	    Optional: (*bool)(<nil>)
            	            	   }),
            	            	   SecretKeyRef: (*v1.SecretKeySelector)(<nil>)
            	            	  })
                  port:
                    type: 0
                    intval: 0
                    strval: ""
                  host: ""
                  scheme: ""
                  httpheaders: []
                tcpsocket: null
                grpc: null
              initialdelayseconds: 1
              timeoutseconds: 1
              periodseconds: 1
              successthreshold: 1
              failurethreshold: 1
              terminationgraceperiodseconds: null
            readinessprobe:
              probehandler:
                exec: null
                httpget:
                  path: /healthz
                  port:
                    type: 0
                    intval: 0
                    strval: ""
                  host: ""
                  scheme: ""
                  httpheaders: []
                tcpsocket: null
                grpc: null
              initialdelayseconds: 1
              timeoutseconds: 1
              periodseconds: 1
              successthreshold: 1
              failurethreshold: 1
              terminationgraceperiodseconds: null
            cachedir: ""
            configvalue: '{"mounts":[{"mountPoint":"abc","options":{"a":"b","c":"d"}}],"targetPath":"/thin/fluid/test/thin-fuse"}'
```

**What you expect to happen:**

**How to reproduce it**

**Additional Information**

--- a/codecov.yml
+++ b/codecov.yml
@@ -8,3 +8,4 @@ coverage:
         target: 60%
 ignore:
   - "**/*_generated.*"
+  - "pkg/utils/testutil"

--- a/pkg/ddc/thin/transform_fuse_test.go
+++ b/pkg/ddc/thin/transform_fuse_test.go
@@ -745,7 +745,7 @@ func TestThinEngine_transformFuseWithDuplicateOptionKey(t1 *testing.T) {
 	})
 }
 
-// Function to sort the value of an environment variable by its name
+// sortEnvVarByName sorts the value of an environment variable by its name
 func sortEnvVarByName(envs []corev1.EnvVar, name string) []corev1.EnvVar {
 	// Search for the environment variable with the given name
 	for i := range envs {
@@ -758,7 +758,7 @@ func sortEnvVarByName(envs []corev1.EnvVar, name string) []corev1.EnvVar {
 	return envs
 }
 
-// Function to sort the value of an environment variable
+// sortEnvVarValue sorts the value of an environment variable
 func sortEnvVarValue(value string) string {
 	// Split the value into key=value pairs
 	pairs := strings.Split(value, ",")

--- a/pkg/ddc/thin/transform_fuse_test.go
+++ b/pkg/ddc/thin/transform_fuse_test.go
@@ -17,9 +17,7 @@
 package thin
 
 import (
-	"fmt"
 	"reflect"
-	"sort"
 	"strings"
 	"testing"
 
@@ -736,47 +734,11 @@ func TestThinEngine_transformFuseWithDuplicateOptionKey(t1 *testing.T) {
 			t1.Errorf("transformFuse() error = %v", err)
 		}
 
-		value.Fuse.Envs = sortEnvVarByName(value.Fuse.Envs, common.ThinFuseOptionEnvKey)
+		value.Fuse.Envs = testutil.SortEnvVarByName(value.Fuse.Envs, common.ThinFuseOptionEnvKey)
 		if !testutil.DeepEqualIgnoringSliceOrder(t1, value.Fuse, wantValue.Fuse) {
 			valueYaml, _ := yaml.Marshal(value.Fuse)
 			wantYaml, _ := yaml.Marshal(wantValue.Fuse)
 			t1.Errorf("transformFuse() \ngot = %v, \nwant = %v", string(valueYaml), string(wantYaml))
 		}
 	})
-}
-
-// sortEnvVarByName sorts the value of an environment variable by its name
-func sortEnvVarByName(envs []corev1.EnvVar, name string) []corev1.EnvVar {
-	// Search for the environment variable with the given name
-	for i := range envs {
-		if envs[i].Name == name {
-			// Sort the value of the environment variable
-			envs[i].Value = sortEnvVarValue(envs[i].Value)
-			return envs
-		}
-	}
-	return envs
-}
-
-// sortEnvVarValue sorts the value of an environment variable
-func sortEnvVarValue(value string) string {
-	// Split the value into key=value pairs
-	pairs := strings.Split(value, ",")
-	kvMap := make(map[string]string, len(pairs))
-
-	// Add the key=value pairs to a map
-	for _, pair := range pairs {
-		kv := strings.Split(pair, "=")
-		if len(kv) == 2 {
-			kvMap[kv[0]] = kv[1]
-		}
-	}
-
-	// Sort the keys and build a new value
-	var sortedPairs []string
-	for key, value := range kvMap {
-		sortedPairs = append(sortedPairs, fmt.Sprintf("%s=%s", key, value))
-	}
-	sort.Strings(sortedPairs)
-	return strings.Join(sortedPairs, ",")
 }

--- a/pkg/ddc/thin/transform_fuse_test.go
+++ b/pkg/ddc/thin/transform_fuse_test.go
@@ -17,7 +17,9 @@
 package thin
 
 import (
+	"fmt"
 	"reflect"
+	"sort"
 	"strings"
 	"testing"
 
@@ -734,10 +736,47 @@ func TestThinEngine_transformFuseWithDuplicateOptionKey(t1 *testing.T) {
 			t1.Errorf("transformFuse() error = %v", err)
 		}
 
+		value.Fuse.Envs = sortEnvVarByName(value.Fuse.Envs, common.ThinFuseOptionEnvKey)
 		if !testutil.DeepEqualIgnoringSliceOrder(t1, value.Fuse, wantValue.Fuse) {
 			valueYaml, _ := yaml.Marshal(value.Fuse)
 			wantYaml, _ := yaml.Marshal(wantValue.Fuse)
 			t1.Errorf("transformFuse() \ngot = %v, \nwant = %v", string(valueYaml), string(wantYaml))
 		}
 	})
+}
+
+// Function to sort the value of an environment variable by its name
+func sortEnvVarByName(envs []corev1.EnvVar, name string) []corev1.EnvVar {
+	// Search for the environment variable with the given name
+	for i := range envs {
+		if envs[i].Name == name {
+			// Sort the value of the environment variable
+			envs[i].Value = sortEnvVarValue(envs[i].Value)
+			return envs
+		}
+	}
+	return envs
+}
+
+// Function to sort the value of an environment variable
+func sortEnvVarValue(value string) string {
+	// Split the value into key=value pairs
+	pairs := strings.Split(value, ",")
+	kvMap := make(map[string]string, len(pairs))
+
+	// Add the key=value pairs to a map
+	for _, pair := range pairs {
+		kv := strings.Split(pair, "=")
+		if len(kv) == 2 {
+			kvMap[kv[0]] = kv[1]
+		}
+	}
+
+	// Sort the keys and build a new value
+	var sortedPairs []string
+	for key, value := range kvMap {
+		sortedPairs = append(sortedPairs, fmt.Sprintf("%s=%s", key, value))
+	}
+	sort.Strings(sortedPairs)
+	return strings.Join(sortedPairs, ",")
 }

--- a/pkg/utils/testutil/envvars.go
+++ b/pkg/utils/testutil/envvars.go
@@ -1,0 +1,45 @@
+package testutil
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// SortEnvVarByName sorts the value of an environment variable by its name
+func SortEnvVarByName(envs []corev1.EnvVar, name string) []corev1.EnvVar {
+	// Search for the environment variable with the given name
+	for i := range envs {
+		if envs[i].Name == name {
+			// Sort the value of the environment variable
+			envs[i].Value = sortEnvVarValue(envs[i].Value)
+			return envs
+		}
+	}
+	return envs
+}
+
+// sortEnvVarValue sorts the value of an environment variable
+func sortEnvVarValue(value string) string {
+	// Split the value into key=value pairs
+	pairs := strings.Split(value, ",")
+	kvMap := make(map[string]string, len(pairs))
+
+	// Add the key=value pairs to a map
+	for _, pair := range pairs {
+		kv := strings.Split(pair, "=")
+		if len(kv) == 2 {
+			kvMap[kv[0]] = kv[1]
+		}
+	}
+
+	// Sort the keys and build a new value
+	var sortedPairs []string
+	for key, value := range kvMap {
+		sortedPairs = append(sortedPairs, fmt.Sprintf("%s=%s", key, value))
+	}
+	sort.Strings(sortedPairs)
+	return strings.Join(sortedPairs, ",")
+}

